src/turfs/processing.rs

```diff
@@ -172,172 +224,207 @@ fn _process_turf_start() -> Result<(), String> {
-					let prev_cost = ssair
-						.get_number(byond_string!("cost_post_process"))
-						.map_err(|_| {
-							runtime!(
-								"Attempt to interpret non-number value as number {} {}:{}",
-								std::file!(),
-								std::line!(),
-								std::column!()
-							)
-						})?;
-					ssair.set(
-						byond_string!("cost_post_process"),
-						Value::from(0.8 * prev_cost + 0.2 * (bench as f32)),
-					)?;
-					Ok(Value::null())
-				}));
-			}
-			{
-				drop(sender.try_send(Box::new(move || {
-					for callback in stats.iter() {
-						callback()?;
-					}
-					Ok(Value::null())
-				})));
-			}
+				});
+			/*			{
+							drop(sender.try_send(Box::new(move || {
+								for callback in stats.iter() {
+									callback()?;
+								}
+								Ok(Value::null())
+							})));
+						}
+			*/
+			let bench = start_time.elapsed().as_millis();
+			sender.try_send(Box::new(move || -> DMResult {
+				let ssair = auxtools::Value::globals().get(byond_string!("SSair"))?;
+				let prev_cost = ssair.get_number(byond_string!("cost_turfs")).map_err(|_| {
+					runtime!(
+						"Attempt to interpret non-number value as number {} {}:{}",
+						std::file!(),
+						std::line!(),
+						std::column!()
+					)
+				})?;
+				ssair.set(
+					byond_string!("cost_turfs"),
+					Value::from(0.8 * prev_cost + 0.2 * (bench as f32)),
+				)?;
+				Ok(Value::null())
+			}));
 			TASKS_RUNNING.fetch_sub(1, Ordering::Release);
 		});
 	});
 	Ok(())
 }
 
+fn planet_process() {
+	with_turf_gases_read(|arena| {
+		GasArena::with_all_mixtures(|all_mixtures| {
+			arena
+				.graph
+				.raw_nodes()
+				.par_iter()
+				.filter_map(|node| {
+					Some((
+						node.weight,
+						node.weight
+							.planetary_atmos
+							.and_then(|id| planetary_atmos().try_get(&id).try_unwrap())?,
+					))
+				})
+				.for_each(|(turf_mix, planet_atmos_entry)| {
+					let planet_atmos = planet_atmos_entry.value();
+					if let Some(gas_read) = all_mixtures
+						.get(turf_mix.mix)
+						.and_then(|lock| lock.try_upgradable_read())
+					{
+						let comparison = gas_read.compare(planet_atmos);
+						if let Some(mut gas) = (comparison > GAS_MIN_MOLES)
+							.then(|| {
+								parking_lot::lock_api::RwLockUpgradableReadGuard::try_upgrade(
+									gas_read,
+								)
+								.ok()
+							})
+							.flatten()
+						{
+							if comparison > 0.1 {
+								gas.multiply(1.0 - GAS_DIFFUSION_CONSTANT);
+								gas.merge(&(planet_atmos * GAS_DIFFUSION_CONSTANT));
+							} else {
+								gas.copy_from_mutable(planet_atmos);
+							}
+						}
+					}
+				})
+		})
+	})
+}
+
 // Compares with neighbors, returning early if any of them are valid.
 fn should_process(
 	index: NodeIndex<usize>,
 	mixture: &TurfMixture,
 	all_mixtures: &[RwLock<Mixture>],
-	arena: &RwLockReadGuard<TurfGases>,
+	arena: &TurfGases,
 ) -> bool {
-	arena.adjacent_infos(index).next().is_some()
-		&& mixture.enabled()
+	mixture.enabled()
+		&& arena.adjacent_infos(index).next().is_some()
```

Comment:
singletons are already filtered out??

Reply:
this is an issue but it's a separate one, one sec

Reply:
wait no i undid having space turfs have no outgoings.